meta-mender-core/recipes-mender/mender/mender_git.bb

```diff
@@ -6,23 +6,30 @@ require mender.inc
 # SHA.
 def mender_autorev_if_git_version(d):
     version = d.getVar("PREFERRED_VERSION")
+    if version is None or version == "":
+        version = d.getVar("PREFERRED_VERSION_%s" % d.getVar('PN'))
     if version is not None and "git" in version:
         return d.getVar("AUTOREV")
     else:
         return "f6ffa190892202263fdb75975059fbb201adab6a"
 SRCREV ?= '${@mender_autorev_if_git_version(d)}'
 
-def mender_branch_from_preferred_version(pref_version):
+def mender_branch_from_preferred_version(d):
     import re
-    match = re.match(r"^[0-9]+\.[0-9]+\.", pref_version)
+    version = d.getVar("PREFERRED_VERSION")
+    if version is None or version == "":
+        version = d.getVar("PREFERRED_VERSION_%s" % d.getVar('PN'))
+    if version is None:
+        version = ""
+    match = re.match(r"^[0-9]+\.[0-9]+\.", version)
     if match is not None:
         # If the preferred version is some kind of version, use the branch name
         # for that one (1.0.x style).
         return match.group(0) + "x"
     else:
         # Else return master as branch.
         return "master"
-MENDER_BRANCH = "${@mender_branch_from_preferred_version('${PREFERRED_VERSION}')}"
+MENDER_BRANCH = "${@mender_branch_from_preferred_version(d)}"
```

Comment:
@drewmoseley: The key is here. If `PREFERRED_VERSION` was unset, it would trigger a parse error, so now we pass the always defined `d` instead, and fetch the variable from there, with proper checking for `None`.

I'm not completely certain why this only started showing up once we made `MENDER_CERT_LOCATION` dependent on a function call. `MENDER_CERT_LOCATION` started depending on `MENDER_BRANCH`, so I suspect it has to do with recursive calls having undefined values. By using `d` directly instead, we avoid the undefined values.